src/components/Notification/Notification.tsx

```diff
@@ -0,0 +1,175 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React, { useEffect, useRef } from "react";
+import type { HTMLProps, ReactNode } from "react";
+
+import Button, { ButtonAppearance } from "../Button";
+import type { Props as ButtonProps } from "../Button";
+
+export const notificationTypes = {
+  CAUTION: "caution",
+  INFORMATION: "information",
+  NEGATIVE: "negative",
+  POSITIVE: "positive",
+};
+
+/**
+ * The props for the Notification component.
+ */
+export type Props = {
+  /**
+   * The list of actions that the notification can perform.
+   */
+  actions?: {
+    appearance?: ButtonProps["appearance"];
+    className?: string;
+    label: string;
+    onClick: () => void;
+  }[];
+  /**
+   * Whether the notification should not show a border.
+   */
+  appearance?: "borderless" | "default" | "modal" | "raised";
+  /**
+   * The notification message content.
+   */
+  children?: ReactNode;
+  /**
+   * Optional class(es) to apply to the parent notification element.
+   */
+  className?: string;
+  /**
+   * Whether the title should display inline with the message.
+   */
+  inline?: boolean;
+  /**
+   * The function to run when dismissing/closing the notification.
+   */
+  onDismiss?: () => void;
+  /**
+   * The severity of the notification.
+   */
+  severity?: "caution" | "information" | "negative" | "positive";
+  /**
+   * The amount of time (in ms) until the notification is automatically dismissed.
+   */
+  timeout?: number;
+  /**
+   * A relevant timestamp for the notification, e.g. when it was created.
+   */
+  timestamp?: ReactNode;
+  /**
+   * The title of the notification.
+   */
+  title?: ReactNode;
+} & HTMLProps<HTMLDivElement>;
+
+const Notification = ({
+  actions,
+  appearance = "default",
+  children,
+  className,
+  inline = false,
+  onDismiss,
+  severity = "information",
+  timeout,
+  timestamp,
+  title,
+  ...props
+}: Props): JSX.Element => {
+  const timeoutId = useRef(null);
+  const hasActions = actions?.length;
+  const showMeta = timestamp || hasActions;
+
+  useEffect(() => {
+    if (timeout && onDismiss) {
+      timeoutId.current = setTimeout(() => onDismiss(), timeout);
+    }
+    return () => clearTimeout(timeoutId.current);
+  }, [onDismiss, timeout]);
+
+  return (
+    <div
+      className={classNames(className, {
+        [`p-notification--${severity}`]: !!severity,
+        "p-notification": !severity,
+        "is-borderless": appearance === "borderless",
+        "is-inline": inline,
+        "is-modal": appearance === "modal",
+        "is-raised": appearance === "raised",
+      })}
+      {...props}
+    >
+      <div className="p-notification__content">
+        {title && (
+          <h5 className="p-notification__title" data-test="notification-title">
+            {title}
+          </h5>
+        )}
+        <p className="p-notification__message">{children}</p>
```

Comment:
The `inline` variant seems to depend on a space being part of the title, without it React just merges the elements together:

<img width="1252" alt="Screenshot 2021-07-13 at 12 53 07" src="https://user-images.githubusercontent.com/83575/125440072-8d2d1c04-b6f6-4d88-a748-b316e63d0f4b.png">

We should include the space `{' '}` between title and message to make sure space doesn't need to be part of content.